src/controllers/mainController.ts

```diff
@@ -865,9 +847,56 @@ export default class MainController implements vscode.Disposable {
      * Called by VS Code when user settings are changed
      * @param ConfigurationChangeEvent event that is fired when config is changed
      */
-    public onDidChangeConfiguration(e: vscode.ConfigurationChangeEvent): void {
+    public async onDidChangeConfiguration(e: vscode.ConfigurationChangeEvent): Promise<void> {
         if (e.affectsConfiguration(Constants.extensionName)) {
-            this._objectExplorerProvider.refresh(undefined);
+            let needsRefresh = false;
+            // user connections is a super set of object explorer connections
+            let userConnections: any[] = this._vscodeWrapper.getConfiguration(Constants.extensionName).get(Constants.connectionsArrayName);
+            let objectExplorerConnections = this._objectExplorerProvider.rootNodeConnections;
+
+            // if a connection(s) was/were manually removed
+            if (objectExplorerConnections.length >= userConnections.length) {
+                let staleConnections = objectExplorerConnections.filter((oeConn) => {
+                    return !userConnections.some((userConn) => Utils.isSameConnection(oeConn, userConn));
+                });
+                // disconnect that/those connection(s) and then
+                // remove its/their credentials from the credential store
+                // and MRU
+                for (let conn of staleConnections) {
+                    let profile = <IConnectionProfile>conn;
+                    if (this.connectionManager.isActiveConnection(conn)) {
+                        const uri = this.connectionManager.getUriForConnection(conn);
+                        await this.connectionManager.disconnect(uri);
+                    }
+                    await this.connectionManager.connectionStore.removeRecentlyUsed(profile);
+                    if (profile.authenticationType === Constants.sqlAuthentication &&
+                        profile.savePassword) {
+                            await this.connectionManager.deleteCredential(profile);
+                        }
+                }
+                // remove them from object explorer
+                await this._objectExplorerProvider.removeStaleConnectionNodes(staleConnections);
+                needsRefresh = true;
+            } else if (userConnections.length > objectExplorerConnections.length) {
```

Comment:
>objectExplorerConnections [](start = 48, length = 25)

and then for this you would just take filter all the userConnections that aren't in objectExplorerConnections and add them

Reply:
that's what it's doing right now

Reply:
Yes, but again you're checking the length first though. If someone manually deleted a connection and added a new one in its place then this logic wouldn't correctly add the new node. 

You should never be using the length to determine if anything changed - that's not always going to be an accurate representation.